Common/src/main/java/vazkii/patchouli/client/book/gui/GuiBookPamphletLanding.java

```diff
@@ -0,0 +1,200 @@
+package vazkii.patchouli.client.book.gui;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.resources.language.I18n;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.network.chat.TranslatableComponent;
+import vazkii.patchouli.client.base.PersistentData;
+import vazkii.patchouli.client.book.BookEntry;
+import vazkii.patchouli.client.book.gui.button.*;
+import vazkii.patchouli.client.gui.GuiAdvancementsExt;
+import vazkii.patchouli.common.book.Book;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static vazkii.patchouli.client.book.gui.GuiBookEntryList.ENTRIES_IN_FIRST_PAGE;
+import static vazkii.patchouli.client.book.gui.GuiBookEntryList.ENTRIES_PER_PAGE;
+
+// A lot of this is a copy of GuiBookLanding
```

Comment:
Hmm. I don't like all this copypasting. What about this?:

Modify GuiBookLanding to directly check if the book only has one category. If so, display the entries of that category inline on the right, else do the existing logic. You can probably factor out the entry button building code from the existing place so you don't duplicate that.

Doing it that way would avoid this new copy class entirely, and avoid all the getLandingPage changes too.

Reply:
It would also avoid the need to have pamphlet_category entirely. Instead, you'd have something like pamphlet_mode: bool to enable this feature (or we could always enable it?)